types/mapbox__mapbox-gl-draw/index.d.ts

```diff
@@ -159,51 +159,65 @@ declare namespace MapboxDraw {
 
-        onMouseUp?(this: DrawCustomModeThis, state: CustomModeState, e: MapMouseEvent): void;
+        onMouseUp?(this: DrawCustomModeThis & this, state: CustomModeState, e: MapMouseEvent): void;
 
-        onMouseOut?(this: DrawCustomModeThis, state: CustomModeState, e: MapMouseEvent): void;
+        onMouseOut?(this: DrawCustomModeThis & this, state: CustomModeState, e: MapMouseEvent): void;
 
-        onKeyUp?(this: DrawCustomModeThis, state: CustomModeState, e: KeyboardEvent): void;
+        onKeyUp?(this: DrawCustomModeThis & this, state: CustomModeState, e: KeyboardEvent): void;
 
-        onKeyDown?(this: DrawCustomModeThis, state: CustomModeState, e: KeyboardEvent): void;
+        onKeyDown?(this: DrawCustomModeThis & this, state: CustomModeState, e: KeyboardEvent): void;
 
-        onTouchStart?(this: DrawCustomModeThis, state: CustomModeState, e: MapTouchEvent): void;
+        onTouchStart?(this: DrawCustomModeThis & this, state: CustomModeState, e: MapTouchEvent): void;
 
-        onTouchMove?(this: DrawCustomModeThis, state: CustomModeState, e: MapTouchEvent): void;
+        onTouchMove?(this: DrawCustomModeThis & this, state: CustomModeState, e: MapTouchEvent): void;
 
-        onTouchEnd?(this: DrawCustomModeThis, state: CustomModeState, e: MapTouchEvent): void;
+        onTouchEnd?(this: DrawCustomModeThis & this, state: CustomModeState, e: MapTouchEvent): void;
 
-        onTap?(this: DrawCustomModeThis, state: CustomModeState, e: MapTouchEvent): void;
+        onTap?(this: DrawCustomModeThis & this, state: CustomModeState, e: MapTouchEvent): void;
 
-        onStop?(this: DrawCustomModeThis, state: CustomModeState): void;
+        onStop?(this: DrawCustomModeThis & this, state: CustomModeState): void;
 
-        onTrash?(this: DrawCustomModeThis, state: CustomModeState): void;
+        onTrash?(this: DrawCustomModeThis & this, state: CustomModeState): void;
 
-        onCombineFeature?(this: DrawCustomModeThis, state: CustomModeState): void;
+        onCombineFeature?(this: DrawCustomModeThis & this, state: CustomModeState): void;
 
-        onUncombineFeature?(this: DrawCustomModeThis, state: CustomModeState): void;
+        onUncombineFeature?(this: DrawCustomModeThis & this, state: CustomModeState): void;
 
         toDisplayFeatures(
-            this: DrawCustomModeThis,
+            this: DrawCustomModeThis & this,
             state: CustomModeState,
             geojson: GeoJSON,
             display: (geojson: GeoJSON) => void,
         ): void;
     }
+
+    interface Modes {
+        [modeKey: string]: DrawCustomMode;
+        draw_line_string: DrawCustomMode;
+        draw_polygon: DrawCustomMode;
+        draw_point: DrawCustomMode;
+        simple_select: DrawCustomMode;
+        direct_select: DrawCustomMode;
+        static: DrawCustomMode;
+    }
+
+    type IMapboxDrawOptions = ConstructorParameters<typeof MapboxDraw>[0];
```

Comment:
It seems like this should be `MapboxDrawOptions`, without the `I` prefix to fit with the rest of this code.

Reply:
Also, this appears to only exist for testing; is this required to add?

Reply:
I wanted to rename it, but it was to late, sorry
It's not 'required', but really nice to have. In tests I add common use case for it

Reply:
Oh, this got merged, I guess I should have clicked the red button for this comment :(